pkg/sr/client.go

```diff
@@ -150,12 +152,13 @@ start:
 			return fmt.Errorf("pre-request hook failed for %s %q: %w", method, reqURL, err)
 		}
 	}
-
+	cl.logger.Log(LogLevelDebug, "sending request", "method", method, "URL", reqURL, "bearer", cl.bearerToken != "", "basicAuth", cl.basicAuth != nil)
```

Comment:
Wdyt about `has_bearer_token` and `has_basic_auth`
Also wdyt about switching on the log level: if info, log without has_{basic|bearer}_stuff; if debug, log with?

Reply:
I like it, I added the `Level()` method to the interface and changed the name to `has_{basic|bearer}`